bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoCoffeeHandler.java

```diff
@@ -272,31 +268,25 @@ protected void updateWemoState() {
                 + action + ">" + "</s:Body>" + "</s:Envelope>";
 
         try {
-            String wemoURL = getWemoURL(actionService);
+            URL descriptorURL = service.getDescriptorURL(this);
+            String wemoURL = getWemoURL(descriptorURL, actionService);
+
             if (wemoURL != null) {
-                String wemoCallResponse = wemoHttpCaller.executeCall(wemoURL, soapHeader, content);
+                String wemoCallResponse = wemoCall.executeCall(wemoURL, soapHeader, content);
                 if (wemoCallResponse != null) {
                     try {
-                        String stringParser = StringUtils.substringBetween(wemoCallResponse, "<attributeList>",
-                                "</attributeList>");
+                        String stringParser = substringBetween(wemoCallResponse, "<attributeList>", "</attributeList>");
 
                         // Due to Belkins bad response formatting, we need to run this twice.
-                        stringParser = StringEscapeUtils.unescapeXml(stringParser);
-                        stringParser = StringEscapeUtils.unescapeXml(stringParser);
+                        stringParser = unescapeXml(stringParser);
+                        stringParser = unescapeXml(stringParser);
 
                         logger.trace("CoffeeMaker response '{}' for device '{}' received", stringParser,
                                 getThing().getUID());
 
                         stringParser = "<data>" + stringParser + "</data>";
 
                         DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
-                        // see
-                        // https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html
-                        dbf.setFeature("http://xml.org/sax/features/external-general-entities", false);
-                        dbf.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
-                        dbf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
-                        dbf.setXIncludeAware(false);
-                        dbf.setExpandEntityReferences(false);
```

Comment:
What is the reason for this change?

Reply:
Introduced with this Security advisory:
https://github.com/openhab/openhab-addons/security/advisories/GHSA-r2hc-pmr7-4c9r

Reply:
My point is that you are removing the fix of https://github.com/openhab/openhab-addons/commit/b0a15b48a3e5c07e4657514c6841cafcaa258175#diff-4780ada0429e132073d2ae9d5605f6fe53a3f398fff40ea05a3869e15b7b2427. I.e. you reintroduce the vulnerability.

Reply:
@wborn readded ...